pennylane/resources/first_quantization.py

```diff
@@ -0,0 +1,221 @@
+        + (7 * 2 ** (n_p + 1) - 9 * n_p - 11 - 3 * 2 ** (-1 * n_p))
+    )
+
+    e_r = _cost_qrom_min(l_z)
+
+    cost = 2 * (n_t + 4 * n_etaz + 2 * br - 12) + 14 * n_eta + 8 * br - 36
+    cost += 3 * n_p**2 + 15 * n_p - 7 + 4 * n_m * (n_p + 1)
+    cost += l_z + e_r + 2 * (2 * n_p + 2 * br - 7) + 12 * eta * n_p
+    cost += 5 * (n_p - 1) + 2 + 24 * n_p + 6 * n_p + n_r + 18
+    cost += n_etaz + 2 * n_eta + 6 * n_p + n_m + 16  # + polylog 1/epsilon
+
+    return int(np.ceil(cost))
+
+
+def estimation_cost(lamb, error):
+    r"""Return the number of calls to the unitary needed to achieve the desired error in quantum
+    phase estimation.
+
+    Args:
+        lamb (float): 1-norm of a second-quantized Hamiltonian
+        error (float): target error in the algorithm
+
+    Returns:
+        int: number of calls to unitary
+
+    **Example**
+
+    >>> cost = estimation_cost(72.49779513025341, 0.001)
+    >>> print(cost)
+    113880
+    """
+    return int(np.ceil(np.pi * lamb / (2 * error)))
+
+
+def gate_cost(n, eta, omega, error, lamb, br=7, charge=0):
+    r"""Return the total number of Toffoli gates needed to implement the first quantization
+    algorithm.
+
+    Args:
+        n (int): number of basis states
+        eta (int): number of electrons
+        omega (float): unit cell volume
+        error (float): target error in the algorithm
+        lamb (float): 1-norm of the Hamiltonian
+        br (int): number of bits for ancilla qubit rotation
+        charge (int): total electric charge of the system
+
+    Returns:
+        int: the number of Toffoli gates needed to implement the first quantization algorithm
+
+    **Example**
+
+    >>> n = 100000
+    >>> eta = 156
+    >>> omega = 169.69608
+    >>> error = 0.01
+    >>> lamb = 5128920.595980267
+    >>> gate_cost(n, eta, omega, error, lamb)
+    10327614069516
+    """
+    e_cost = estimation_cost(lamb, error)
+    u_cost = unitary_cost(n, eta, omega, error, lamb, br, charge)
+
+    return e_cost * u_cost
+
+
+def qubit_cost(n, eta, omega, error, lamb, charge=0):
+    r"""Return the number of ancilla qubits needed to implement the first quantization algorithm.
+    ￼
+    The expression for computing the cost is taken from
+    [`arXiv:arXiv:2204.11890 <https://arxiv.org/abs/2204.11890>`_].
+
+    Args:
+        n (int): number of basis states
+        eta (int): number of electrons
+        omega (float): unit cell volume
+        error (float): target error in the algorithm
+        lamb (float): 1-norm of the Hamiltonian
+        charge (int): total electric charge of the system
+
+    Returns:
+        int: the number of ancilla qubits needed to implement the first quantization algorithm
+
+    **Example**
+
+    >>> n = 100000
+    >>> eta = 156
+    >>> omega = 169.69608
+    >>> error = 0.01
+    >>> lamb = 5128920.595980267
+    >>> qubit_cost(n, eta, omega, error, lamb)
+    4238
+    """
+    l_z = eta + charge
+    l_nu = 4 * np.pi * n ** (1 / 3)
```

Comment:
Shouldn't it be 
```python3
l_nu = 2 * np.pi * n ** (2 / 3)
```
(based on `integrate(r*sin(t), (r, 0, N), (p, 0, 2*pi), (t, 0, pi))` and eq 13 in [here](https://www.nature.com/articles/s41534-019-0199-y) )

Reply:
Not sure myself, couldn't find the Eq. for lambda_nu after a quick search. If a change is needed, then it should be applied also in the previous function